IABot/Parser/parse.php

```diff
@@ -487,18 +490,17 @@ public function updateLinkInfo( $links ) {
 		foreach( $links as $tid => $link ) {
 			if( ( $this->commObject->TOUCH_ARCHIVE == 1 || $link['has_archive'] === false ) && $this->commObject->VERIFY_DEAD == 1 && $this->commObject->db->dbValues[$tid]['live_state'] !== 0 && $this->commObject->db->dbValues[$tid]['live_state'] !== 5 && (time() - $this->commObject->db->dbValues[$tid]['last_deadCheck'] > 259200) ) $toCheck[$tid] = $link['url'];
 		}
-		$results = $this->deadCheck->checkDeadlinks( $toCheck );
-		$results = $results['results'];
+		$results = $this->deadCheck->areLinksDead( $toCheck );
 		foreach( $links as $tid => $link ) {
 			$link['is_dead'] = null;
 			if( ( $this->commObject->TOUCH_ARCHIVE == 1 || $link['has_archive'] === false ) && $this->commObject->VERIFY_DEAD == 1 ) {
 				if( $this->commObject->db->dbValues[$tid]['live_state'] != 0 && $this->commObject->db->dbValues[$tid]['live_state'] != 5 && (time() - $this->commObject->db->dbValues[$tid]['last_deadCheck'] > 259200) ) {
-					$link['is_dead'] = $results[$tid];
-					$this->commObject->db->dbValues[$tid]['last_deadCheck'] = time(); 
+					$link['is_dead'] = $results[$link['url']];
```

Comment:
Is the URL now the array index?   I had programmed the checkIfDead to preserve array index keys when returning data.